From ed83019ac653bff197aed9e453603832db276982 Mon Sep 17 00:00:00 2001
From: "Brett T. Warden" <brett.t.warden@intel.com>
Date: Wed, 13 Jan 2021 16:32:25 -0800
Subject: [PATCH] Fix rpc_pipefs paths to /run/nfs

We set the NFS state dir to /run/nfs. Some programs that use the
rpc_pipefs filesystem properly follow the configured path. Others have
/var/lib/nfs hard-coded.

Update remaining programs to use the NFS_STATEDIR definition, manually
edit the man pages to the full path, rewrite the systemd unit files, and
correct the remaining makefiles to reflect that rpc_pipefs will be
mounted under /run/nfs/, not /var/lib/nfs/.
---
 nfs.conf                                                      | 2 +-
 systemd/Makefile.am                                           | 2 +-
 systemd/rpc-pipefs-generator.c                                | 2 +-
 systemd/rpc_pipefs.target                                     | 4 ++--
 ...{var-lib-nfs-rpc_pipefs.mount => run-nfs-rpc_pipefs.mount} | 2 +-
 utils/blkmapd/blkmapd.man                                     | 2 +-
 utils/blkmapd/device-discovery.c                              | 2 +-
 utils/gssd/gssd.h                                             | 2 +-
 utils/gssd/gssd.man                                           | 2 +-
 utils/idmapd/idmapd.c                                         | 2 +-
 utils/idmapd/idmapd.man                                       | 2 +-
 utils/nfsdcld/nfsdcld.man                                     | 2 +-
 12 files changed, 13 insertions(+), 13 deletions(-)
 rename systemd/{var-lib-nfs-rpc_pipefs.mount => run-nfs-rpc_pipefs.mount} (84%)

diff --git a/nfs.conf b/nfs.conf
index 186a5b19e24c..e7ef672eb9c8 100644
--- a/nfs.conf
+++ b/nfs.conf
@@ -3,7 +3,7 @@
 # NFS daemons and tools
 #
 [general]
-# pipefs-directory=/var/lib/nfs/rpc_pipefs
+# pipefs-directory=/run/nfs/rpc_pipefs
 #
 [exports]
 # rootdir=/export
diff --git a/systemd/Makefile.am b/systemd/Makefile.am
index 75cdd9f54c81..bb7f3e2c8e46 100644
--- a/systemd/Makefile.am
+++ b/systemd/Makefile.am
@@ -13,7 +13,7 @@ unit_files =  \
     rpc-statd.service \
     \
     proc-fs-nfsd.mount \
-    var-lib-nfs-rpc_pipefs.mount
+    run-nfs-rpc_pipefs.mount
 
 if CONFIG_NFSV4
 unit_files += \
diff --git a/systemd/rpc-pipefs-generator.c b/systemd/rpc-pipefs-generator.c
index 8e218aa70645..c24db567a45d 100644
--- a/systemd/rpc-pipefs-generator.c
+++ b/systemd/rpc-pipefs-generator.c
@@ -21,7 +21,7 @@
 #include "conffile.h"
 #include "systemd.h"
 
-#define RPC_PIPEFS_DEFAULT "/var/lib/nfs/rpc_pipefs"
+#define RPC_PIPEFS_DEFAULT NFS_STATEDIR "/rpc_pipefs"
 
 static int generate_mount_unit(const char *pipefs_path, const char *pipefs_unit,
 			       const char *dirname)
diff --git a/systemd/rpc_pipefs.target b/systemd/rpc_pipefs.target
index 01d4d2786982..d56d969de8df 100644
--- a/systemd/rpc_pipefs.target
+++ b/systemd/rpc_pipefs.target
@@ -1,3 +1,3 @@
 [Unit]
-Requires=var-lib-nfs-rpc_pipefs.mount
-After=var-lib-nfs-rpc_pipefs.mount
+Requires=run-nfs-rpc_pipefs.mount
+After=run-nfs-rpc_pipefs.mount
diff --git a/systemd/var-lib-nfs-rpc_pipefs.mount b/systemd/run-nfs-rpc_pipefs.mount
similarity index 84%
rename from systemd/var-lib-nfs-rpc_pipefs.mount
rename to systemd/run-nfs-rpc_pipefs.mount
index 26d1c7634057..23342fd1a59e 100644
--- a/systemd/var-lib-nfs-rpc_pipefs.mount
+++ b/systemd/run-nfs-rpc_pipefs.mount
@@ -6,5 +6,5 @@ Conflicts=umount.target
 
 [Mount]
 What=sunrpc
-Where=/var/lib/nfs/rpc_pipefs
+Where=/run/nfs/rpc_pipefs
 Type=rpc_pipefs
diff --git a/utils/blkmapd/blkmapd.man b/utils/blkmapd/blkmapd.man
index 4b3d3f04e62a..6cfea23e71ae 100644
--- a/utils/blkmapd/blkmapd.man
+++ b/utils/blkmapd/blkmapd.man
@@ -56,7 +56,7 @@ configuration file:
 Tells
 .B blkmapd
 where to look for the rpc_pipefs filesystem.  The default value is
-.IR /var/lib/nfs/rpc_pipefs .
+.IR /run/nfs/rpc_pipefs .
 .SH SEE ALSO
 .BR nfs (5),
 .BR dmsetup (8),
diff --git a/utils/blkmapd/device-discovery.c b/utils/blkmapd/device-discovery.c
index f5f9b10b95f2..a7ab895c2b85 100644
--- a/utils/blkmapd/device-discovery.c
+++ b/utils/blkmapd/device-discovery.c
@@ -63,7 +63,7 @@
 #define EVENT_SIZE (sizeof(struct inotify_event))
 #define EVENT_BUFSIZE (1024 * EVENT_SIZE)
 
-#define RPCPIPE_DIR	"/var/lib/nfs/rpc_pipefs"
+#define RPCPIPE_DIR	NFS_STATEDIR "/rpc_pipefs"
 #define PID_FILE	"/var/run/blkmapd.pid"
 
 #define CONF_SAVE(w, f) do {			\
diff --git a/utils/gssd/gssd.h b/utils/gssd/gssd.h
index 1e8c58d47203..9fc5ef2079e6 100644
--- a/utils/gssd/gssd.h
+++ b/utils/gssd/gssd.h
@@ -39,7 +39,7 @@
 #include <pthread.h>
 
 #ifndef GSSD_PIPEFS_DIR
-#define GSSD_PIPEFS_DIR		"/var/lib/nfs/rpc_pipefs"
+#define GSSD_PIPEFS_DIR		NFS_STATEDIR "/rpc_pipefs"
 #endif
 #define DNOTIFY_SIGNAL		(SIGRTMIN + 3)
 
diff --git a/utils/gssd/gssd.man b/utils/gssd/gssd.man
index 26095a898293..b967af5284fc 100644
--- a/utils/gssd/gssd.man
+++ b/utils/gssd/gssd.man
@@ -233,7 +233,7 @@ supports settable encryption types.
 Tells
 .B rpc.gssd
 where to look for the rpc_pipefs filesystem.  The default value is
-.IR /var/lib/nfs/rpc_pipefs .
+.IR /run/nfs/rpc_pipefs .
 .TP
 .BI "-d " search-path
 This option specifies a colon separated list of directories that
diff --git a/utils/idmapd/idmapd.c b/utils/idmapd/idmapd.c
index 51c71fbbc643..e2c160e8a4fd 100644
--- a/utils/idmapd/idmapd.c
+++ b/utils/idmapd/idmapd.c
@@ -73,7 +73,7 @@
 #include "nfslib.h"
 
 #ifndef PIPEFS_DIR
-#define PIPEFS_DIR  "/var/lib/nfs/rpc_pipefs/"
+#define PIPEFS_DIR  NFS_STATEDIR "/rpc_pipefs/"
 #endif
 
 #ifndef NFSD_DIR
diff --git a/utils/idmapd/idmapd.man b/utils/idmapd/idmapd.man
index 5f34d2bff860..bcdfec73f60c 100644
--- a/utils/idmapd/idmapd.man
+++ b/utils/idmapd/idmapd.man
@@ -69,7 +69,7 @@ in the foreground and prints all output to the terminal.
 .It Fl p Ar path
 Specifies the location of the RPC pipefs to be
 .Ar path .
-The default value is \&"/var/lib/nfs/rpc_pipefs\&".
+The default value is \&"/run/nfs/rpc_pipefs\&".
 .It Fl c Ar path
 Use configuration file
 .Ar path .
diff --git a/utils/nfsdcld/nfsdcld.man b/utils/nfsdcld/nfsdcld.man
index 861f1c49efec..167ea809b982 100644
--- a/utils/nfsdcld/nfsdcld.man
+++ b/utils/nfsdcld/nfsdcld.man
@@ -158,7 +158,7 @@ Runs the daemon in the foreground and prints all output to stderr
 .IP "\fB\-p\fR \fIpath\fR, \fB\-\-pipefsdir\fR=\fIpath\fR" 4
 .IX Item "-p path, --pipefsdir=path"
 Location of the rpc_pipefs filesystem. The default value is
-\&\fI/var/lib/nfs/rpc_pipefs\fR.
+\&\fI/run/nfs/rpc_pipefs\fR.
 .IP "\fB\-s\fR \fIstorage_dir\fR, \fB\-\-storagedir\fR=\fIstorage_dir\fR" 4
 .IX Item "-s storagedir, --storagedir=storage_dir"
 Directory where stable storage information should be kept. The default
-- 
2.30.0

